new-pipeline/src/evo.js

```diff
@@ -412,8 +412,14 @@ function updateAggregates(kind, buckets) {
   }
 
   // Load aggregates from state on first load
-  newAggregates = newAggregates.map(entry => entry[0]);
-  let aggregates = gInitialPageState.aggregates.filter(aggregate => newAggregates.includes(aggregate));
+  let aggregates = gInitialPageState.aggregates
+                    .filter((aggregate)=>{
+                      return newAggregates.filter((newaggregate)=> newaggregate[1] == aggregate).length > 0
```

Comment:
This code filters based on aggregates who are present as the second index in one of the newAggregates? Perhaps this will be more clear?

`.filter(aggregate => newAggregates.some(newAggregate => newAggregate[1] === aggregate))`

Reply:
The transformation happening here is actually from the values to keys because as we are now storing the values of aggregates in the url instead of the keys so while restoring data back into to UI i have remapped the values to the keys for the multi-select control to work as required.

Reply:
And you are right at the 0th index the keys & at the 1st index the values are stored respectively.